annotations/src/main/java/com/squareup/anvil/annotations/ContributesBindingToMap.kt

```diff
@@ -0,0 +1,14 @@
+package com.squareup.anvil.annotations
+
+import kotlin.annotation.AnnotationRetention.RUNTIME
+import kotlin.annotation.AnnotationTarget.CLASS
+import kotlin.reflect.KClass
+
+@Target(CLASS)
+@Retention(RUNTIME)
+annotation class ContributesBindingToMap(
+  val scope: KClass<*>,
+  val key: KClass<*>,
```

Comment:
@vRallev, would it make sense to you to allow an arbitrary annotation here? It might lose a little in the way of type specificity, but validation wouldn't be too hard in the plugin I don't think. The plugin would just need to make sure the annotation meets dagger qualifier requirements (maybe those are as simple as "has at least one property").